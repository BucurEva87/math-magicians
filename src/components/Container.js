import React from 'react';
import { Routes, Route } from 'react-router-dom';
import Quote from './Quote';
import CalculatorContainer from './CalculatorContainer';
import Home from './Home';

import '../styles/Container.scss';

const Container = () => (
  <div id="container">
    <Routes>
      <Route path="/quote" element={<Quote />} />
      <Route path="/calculator" element={<CalculatorContainer />} />
      <Route path="/" element={<Home />} />
    </Routes>
  </div>
);

export default Container;
