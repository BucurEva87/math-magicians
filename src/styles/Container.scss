#container {
  flex-grow: 1;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 2rem;
}
