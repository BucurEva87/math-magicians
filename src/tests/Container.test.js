import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Container from '../components/Container';

let getByTestId;

beforeEach(() => {
  const component = render(<Router><Container /></Router>);
  getByTestId = component.getByTestId;
});

describe('Container component', () => {
  test('renders on the screen', () => {
    expect(getByTestId('container')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Router><Container /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
