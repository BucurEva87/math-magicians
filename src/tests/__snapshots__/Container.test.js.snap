// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component matches snapshot 1`] = `
<div
  data-testid="container"
  id="container"
>
  <section
    className="container-section"
    data-testid="home-section"
  >
    <h1
      data-testid="home-h1"
    >
      Welcome to our page!
    </h1>
    <p
      data-testid="home-p-first"
    >
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Earum esse quos voluptate sit tenetur aliquam placeat. In voluptas dolorem et voluptatibus aut. Error provident magni doloremque suscipit dignissimos iusto incidunt repellendus temporibus tempore, quidem ducimus laudantium repudiandae animi tenetur, id odio aliquam! Ipsa, veniam! Officia assumenda facilis ut odit eum eligendi in quod quasi! Lorem ipsum, dolor sit amet consectetur adipisicing elit. Earum esse quos voluptate sit tenetur aliquam placeat. In voluptas dolorem et voluptatibus aut. Error provident magni doloremque suscipit dignissimos iusto incidunt repellendus temporibus tempore, quidem ducimus laudantium repudiandae animi tenetur, id odio aliquam! Ipsa, veniam! Officia assumenda facilis ut odit eum eligendi in quod quasi!
    </p>
    <p
      data-testid="home-p-second"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis, eius. Accusamus repellat voluptates enim qui autem quidem itaque ad ut veniam, facilis, corrupti commodi atque quisquam. Ducimus dolorum asperiores numquam ex ullam dolores, aliquam at esse ut quibusdam pariatur possimus consequuntur exercitationem tempore, voluptatibus in laudantium inventore autem quos qui dolorem nam, quis temporibus! Distinctio, dolor officiis excepturi eligendi eos deleniti natus. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis, eius. Accusamus repellat voluptates enim qui autem quidem itaque ad ut veniam, facilis, corrupti commodi atque quisquam. Ducimus dolorum asperiores numquam ex ullam dolores, aliquam at esse ut quibusdam pariatur possimus consequuntur exercitationem tempore, voluptatibus in laudantium inventore autem quos qui dolorem nam, quis temporibus! Distinctio, dolor officiis excepturi eligendi eos deleniti natus. Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis, eius. Accusamus repellat voluptates enim qui autem quidem itaque ad ut veniam, facilis, corrupti commodi atque quisquam. Ducimus dolorum asperiores numquam ex ullam dolores, aliquam at esse ut quibusdam pariatur possimus consequuntur exercitationem tempore, voluptatibus in laudantium inventore autem quos qui dolorem nam, quis temporibus! Distinctio, dolor officiis excepturi eligendi eos deleniti natus.
    </p>
  </section>
</div>
`;
